From 8a9b4e1a3304374260a37a92e903be277e65c324 Mon Sep 17 00:00:00 2001
From: Ryan McCabe <rmccabe@redhat.com>
Date: Mon, 28 Jul 2014 08:44:57 -0400
Subject: [PATCH] luci: Add the ssh_options attribute to applicable fence
 agents

Support the newly added ssh_options attributes for the fence agents
that will accept it.

Resolves: rhbz#1117398
Signed-off-by: Ryan McCabe <rmccabe@redhat.com>
---
 luci/templates/fence_devices.html   | 72 +++++++++++++++++++++----------
 luci/templates/fence_instances.html | 84 +++++++++++++++++++++++++++++++++++++
 luci/validation/validate_fence.py   | 21 ++++++++--
 3 files changed, 152 insertions(+), 25 deletions(-)

diff --git a/luci/templates/fence_devices.html b/luci/templates/fence_devices.html
index 20dc555..38a82d2 100644
--- a/luci/templates/fence_devices.html
+++ b/luci/templates/fence_devices.html
@@ -504,28 +504,6 @@
       </td>
     </tr>
     <tr>
-      <td>SSH</td>
-      <td>
-        <input type="checkbox" class="checkbox" name="secure"
-          py:attrs="cur_fencedev and cur_fencedev.getAttribute('secure') and {'checked': 'checked'}"/>
-        <label class="choice">Use SSH</label>
-      </td>
-    </tr>
-    <tr>
-      <td>Path to SSH Identity File</td>
-      <td>
-        <input type="text" class="text" name="identity_file"
-          py:attrs="cur_fencedev and {'value': cur_fencedev.getAttribute('identity_file')} or {}"/>
-      </td>
-    </tr>
-    <tr>
-      <td>SSH Options</td>
-      <td>
-        <input type="text" class="text" name="ssh_options"
-          py:attrs="cur_fencedev and {'value': cur_fencedev.getAttribute('ssh_options')} or {}"/>
-      </td>
-    </tr>
-    <tr>
       <td>Force Command Prompt</td>
       <td>
         <input type="text" class="text" name="cmd_prompt"
@@ -2606,6 +2584,13 @@
       </td>
     </tr>
     <tr>
+      <td>SSH Options</td>
+      <td>
+        <input type="text" class="text" name="ssh_options"
+          py:attrs="cur_fencedev and {'value': cur_fencedev.getAttribute('ssh_options')} or {}"/>
+      </td>
+    </tr>
+    <tr>
       <td>Path to SSH Identity File</td>
       <td>
         <input type="text" class="text" name="identity_file"
@@ -2718,6 +2703,28 @@
       </td>
     </tr>
     <tr>
+      <td>SSH</td>
+      <td>
+        <input type="checkbox" class="checkbox" name="secure"
+          py:attrs="cur_fencedev and cur_fencedev.getAttribute('secure') and {'checked': 'checked'}"/>
+        <label class="choice">Use SSH</label>
+      </td>
+    </tr>
+    <tr>
+      <td>SSH Options</td>
+      <td>
+        <input type="text" class="text" name="ssh_options"
+          py:attrs="cur_fencedev and {'value': cur_fencedev.getAttribute('ssh_options')} or {}"/>
+      </td>
+    </tr>
+    <tr>
+      <td>Path to SSH Identity File</td>
+      <td>
+        <input type="text" class="text" name="identity_file"
+          py:attrs="cur_fencedev and {'value': cur_fencedev.getAttribute('identity_file')} or {}"/>
+      </td>
+    </tr>
+    <tr>
       <td>TCP Port</td>
       <td>
         <input type="text" class="text" name="telnet_port"
@@ -3239,6 +3246,13 @@
       </td>
     </tr>
     <tr>
+      <td>SSH Options</td>
+      <td>
+        <input type="text" class="text" name="ssh_options"
+          py:attrs="cur_fencedev and {'value': cur_fencedev.getAttribute('ssh_options')} or {}"/>
+      </td>
+    </tr>
+    <tr>
       <td>Path to SSH Identity File</td>
       <td>
         <input type="text" class="text" name="identity_file"
@@ -3683,6 +3697,13 @@
       </td>
     </tr>
     <tr>
+      <td>SSH Options</td>
+      <td>
+        <input type="text" class="text" name="ssh_options"
+          py:attrs="cur_fencedev and {'value': cur_fencedev.getAttribute('ssh_options')} or {}"/>
+      </td>
+    </tr>
+    <tr>
       <td>Path to SSH Identity File</td>
       <td>
         <input type="text" class="text" name="identity_file"
@@ -4068,6 +4089,13 @@
       </td>
     </tr>
     <tr>
+      <td>SSH Options</td>
+      <td>
+        <input type="text" class="text" name="ssh_options"
+          py:attrs="cur_fencedev and {'value': cur_fencedev.getAttribute('ssh_options')} or {}"/>
+      </td>
+    </tr>
+    <tr>
       <td>Path to SSH Identity File</td>
       <td>
         <input type="text" class="text" name="identity_file"
diff --git a/luci/templates/fence_instances.html b/luci/templates/fence_instances.html
index a82e830..5ef4602 100644
--- a/luci/templates/fence_instances.html
+++ b/luci/templates/fence_instances.html
@@ -48,6 +48,13 @@
               py:attrs="cur_fence_inst and cur_fence_inst.getAttribute('secure') and {'checked': 'checked'}"/>
           </td>
          </tr>
+    <tr>
+      <td>SSH Options</td>
+      <td>
+        <input type="text" class="text" name="ssh_options"
+          py:attrs="cur_fence_inst and {'value': cur_fence_inst.getAttribute('ssh_options')}"/>
+      </td>
+    </tr>
          <tr>
           <td>Path to SSH Identity File</td>
           <td>
@@ -132,6 +139,20 @@
       </td>
     </tr>
     <tr>
+	  <td><span title="Enable SSH operation">Use SSH</span></td>
+      <td>
+        <input type="checkbox" class="checkbox" name="secure"
+          py:attrs="cur_fence_inst and cur_fence_inst.getAttribute('secure') and {'checked': 'checked'}"/>
+      </td>
+    </tr>
+    <tr>
+      <td>SSH Options</td>
+      <td>
+        <input type="text" class="text" name="ssh_options"
+          py:attrs="cur_fence_inst and {'value': cur_fence_inst.getAttribute('ssh_options')}"/>
+      </td>
+    </tr>
+    <tr>
       <td>Path to SSH Identity File</td>
       <td>
         <input type="text" class="text" name="identity_file"
@@ -177,6 +198,13 @@
               py:attrs="cur_fence_inst and cur_fence_inst.getAttribute('secure') and {'checked': 'checked'}"/>
           </td>
          </tr>
+    <tr>
+      <td>SSH Options</td>
+      <td>
+        <input type="text" class="text" name="ssh_options"
+          py:attrs="cur_fence_inst and {'value': cur_fence_inst.getAttribute('ssh_options')}"/>
+      </td>
+    </tr>
          <tr>
           <td>Path to SSH Identity File</td>
           <td>
@@ -266,6 +294,13 @@
               py:attrs="cur_fence_inst and cur_fence_inst.getAttribute('secure') and {'checked': 'checked'}"/>
           </td>
          </tr>
+    <tr>
+      <td>SSH Options</td>
+      <td>
+        <input type="text" class="text" name="ssh_options"
+          py:attrs="cur_fence_inst and {'value': cur_fence_inst.getAttribute('ssh_options')}"/>
+      </td>
+    </tr>
          <tr>
           <td>Path to SSH Identity File</td>
           <td>
@@ -314,6 +349,13 @@
               py:attrs="cur_fence_inst and cur_fence_inst.getAttribute('secure') and {'checked': 'checked'}"/>
           </td>
          </tr>
+    <tr>
+      <td>SSH Options</td>
+      <td>
+        <input type="text" class="text" name="ssh_options"
+          py:attrs="cur_fence_inst and {'value': cur_fence_inst.getAttribute('ssh_options')}"/>
+      </td>
+    </tr>
          <tr>
           <td>Path to SSH Identity File</td>
           <td>
@@ -375,6 +417,27 @@
       </td>
     </tr>
     <tr>
+	  <td><span title="Enable SSH operation">Use SSH</span></td>
+      <td>
+        <input type="checkbox" class="checkbox" name="secure"
+          py:attrs="cur_fence_inst and cur_fence_inst.getAttribute('secure') and {'checked': 'checked'}"/>
+      </td>
+    </tr>
+    <tr>
+      <td>SSH Options</td>
+      <td>
+        <input type="text" class="text" name="ssh_options"
+          py:attrs="cur_fence_inst and {'value': cur_fence_inst.getAttribute('ssh_options')}"/>
+      </td>
+    </tr>
+    <tr>
+      <td>Path to SSH Identity File</td>
+      <td>
+        <input type="text" class="text" name="identity_file"
+          py:attrs="cur_fence_inst and {'value': cur_fence_inst.getAttribute('identity_file')}"/>
+      </td>
+    </tr>
+    <tr>
       <td>Unfencing</td>
       <td>
         <input type="checkbox" class="checkbox" name="unfencing"
@@ -615,6 +678,13 @@
               py:attrs="cur_fence_inst and cur_fence_inst.getAttribute('secure') and {'checked': 'checked'}"/>
           </td>
          </tr>
+    <tr>
+      <td>SSH Options</td>
+      <td>
+        <input type="text" class="text" name="ssh_options"
+          py:attrs="cur_fence_inst and {'value': cur_fence_inst.getAttribute('ssh_options')}"/>
+      </td>
+    </tr>
          <tr>
           <td>Path to SSH Identity File</td>
           <td>
@@ -681,6 +751,13 @@
               py:attrs="cur_fence_inst and cur_fence_inst.getAttribute('secure') and {'checked': 'checked'}"/>
           </td>
          </tr>
+    <tr>
+      <td>SSH Options</td>
+      <td>
+        <input type="text" class="text" name="ssh_options"
+          py:attrs="cur_fence_inst and {'value': cur_fence_inst.getAttribute('ssh_options')}"/>
+      </td>
+    </tr>
          <tr>
           <td>Path to SSH Identity File</td>
           <td>
@@ -1177,6 +1254,13 @@
               py:attrs="cur_fence_inst and cur_fence_inst.getAttribute('secure') and {'checked': 'checked'}"/>
           </td>
          </tr>
+    <tr>
+      <td>SSH Options</td>
+      <td>
+        <input type="text" class="text" name="ssh_options"
+          py:attrs="cur_fence_inst and {'value': cur_fence_inst.getAttribute('ssh_options')}"/>
+      </td>
+    </tr>
          <tr>
           <td>Path to SSH Identity File</td>
           <td>
diff --git a/luci/validation/validate_fence.py b/luci/validation/validate_fence.py
index 0b0fbd0..6e16ea7 100644
--- a/luci/validation/validate_fence.py
+++ b/luci/validation/validate_fence.py
@@ -198,9 +198,6 @@ def val_brocade_fd(fencedev, fence_name, **kw):
 		('login', True),
 		('passwd', False),
 		('passwd_script', False),
-		('secure', False),
-		('identity_file', False),
-		('ssh_options', False),
 		('force_ip_family', False),
 		('cmd_prompt', False),
 		('power_wait', False),
@@ -401,6 +398,7 @@ def val_rsa_fd(fencedev, fence_name, **kw):
 		('ipport', False),
 		('login', True),
 		('secure', False),
+		('ssh_options', False),
 		('identity_file', False),
 		('passwd', False),
 		('passwd_script', False),
@@ -430,6 +428,9 @@ def val_rsb_fd(fencedev, fence_name, **kw):
 		('shell_timeout', False),
 		('login_timeout', False),
 		('retry_on', False),
+		('secure', False),
+		('ssh_options', False),
+		('identity_file', False),
 	)
 
 	errors = config_fence_attr(params, fencedev, fence_name, **kw)
@@ -456,6 +457,7 @@ def val_drac5_fd(fencedev, fence_name, **kw):
 		('module_name', False),
 		('cmd_prompt', False),
 		('secure', False),
+		('ssh_options', False),
 		('identity_file', False),
 		('passwd', False),
 		('passwd_script', False),
@@ -787,6 +789,7 @@ def val_ilo_mp_fd(fencedev, fence_name, **kw):
 		('passwd', False),
 		('passwd_script', False),
 		('secure', False),
+		('ssh_options', False),
 		('identity_file', False),
 		('cmd_prompt', False),
 		('power_wait', False),
@@ -997,6 +1000,7 @@ def val_apc_fi(fenceinst, parent_name, **kw):
 		('port', True),
 		('switch', False),
 		('secure', False),
+		('ssh_options', False),
 		('identity_file', False),
 		('delay', False),
 	)
@@ -1008,6 +1012,7 @@ def val_wti_fi(fenceinst, parent_name, **kw):
 	params = (
 		('port', True),
 		('secure', False),
+		('ssh_options', False),
 		('identity_file', False),
 		('delay', False),
 	)
@@ -1019,6 +1024,7 @@ def val_virsh_fi(fenceinst, parent_name, **kw):
 	params = (
 		('port', True),
 		('secure', False),
+		('ssh_options', False),
 		('identity_file', False),
 		('delay', False),
 	)
@@ -1030,6 +1036,9 @@ def val_brocade_fi(fenceinst, parent_name, **kw):
 	params = (
 		('port', True),
 		('delay', False),
+		('secure', False),
+		('identity_file', False),
+		('ssh_options', False),
 	)
 
 	errors = config_fence_attr(params, fenceinst, parent_name, **kw)
@@ -1065,6 +1074,7 @@ def val_bladecenter_fi(fenceinst, parent_name, **kw):
 		('port', True),
 		('switch', False),
 		('secure', False),
+		('ssh_options', False),
 		('identity_file', False),
 		('delay', False),
 	)
@@ -1077,6 +1087,7 @@ def val_ipdu_fi(fenceinst, parent_name, **kw):
 		('port', True),
 		('switch', False),
 		('secure', False),
+		('ssh_options', False),
 		('identity_file', False),
 		('delay', False),
 	)
@@ -1107,6 +1118,7 @@ def val_lpar_fi(fenceinst, parent_name, **kw):
 		('partition', True),
 		('managed', True),
 		('secure', True),
+		('ssh_options', False),
 		('identity_file', False),
 		('delay', False),
 	)
@@ -1157,6 +1169,7 @@ def val_vmware_fi(fenceinst, parent_name, **kw):
 	params = (
 		('port', True),
 		('secure', False),
+		('ssh_options', False),
 		('identity_file', False),
 		('delay', False),
 	)
@@ -1176,6 +1189,7 @@ def val_ldom_fi(fenceinst, parent_name, **kw):
 	params = (
 		('port', True),
 		('secure', False),
+		('ssh_options', False),
 		('identity_file', False),
 		('delay', False),
 	)
@@ -1223,6 +1237,7 @@ def val_hpblade_fi(fenceinst, parent_name, **kw):
 	params = (
 		('port', True),
 		('secure', False),
+		('ssh_options', False),
 		('identity_file', False),
 		('delay', False),
 	)
-- 
1.9.3

